apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.RetryConformanceTests/RetryConformanceTest.cs

```diff
@@ -104,9 +107,9 @@ private async Task RunTestCase(InstructionList instructionList, Method method, b
                 }
                 catch (Exception ex) // To catch expected exception when retry should not happen.
                 {
-                    if (ex != null && ex is GoogleApiException exception)
+                    if (ex is GoogleApiException)
                     {
-                        var statusCode = exception.HttpStatusCode;
+                        var statusCode = ((GoogleApiException) ex).HttpStatusCode;
```

Comment:
I don't know why you've backed out the pattern matching here. It looked fine to me - I was just suggesting that you didn't need the first nullity check:

```csharp
if (ex is GoogleApiException exception)
{
    var statusCode = exception.StatusCode
    ...
}
```

Reply:
Should I revert it back? This will not be ok?

Reply:
Yes, please change it to my suggestion as per the comment. The code you've got at the moment will work, but when we've already identified simpler code it would seem odd not to use it.